### PR TITLE
chore(examples): Add 'Sitefinity CMS' Example

### PR DESCRIPTION
**Pull request by Gebov**
This is an example of how to integrate [Sitefinity CMS ](https://www.progress.com/sitefinity-cms)with next.js

### CODE REVIEW

File: examples/cms-sitefinity/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+This is the existing [blog-starter](https://github.com/vercel/next.js/tree/canary/examples/blog-starter) plus TypeScript.
+This example showcases [Next.js's Static Generation feature](https://nextjs.org/docs/basic-features/pages) using Sitefinity CMS as the data source.
+
+## Demo
+
+[https://next-blog-starter.vercel.app/](https://next-blog-starter.vercel.app/)
+
+## Deploy your own
+
+Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example) or preview live with [StackBlitz](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/blog-starter)
```
Comment (balazsorban44) [inline examples/cms-sitefinity/README.md:12]:
```suggestion
Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example) or preview live with [StackBlitz](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/cms-sitefinity)
```

File: examples/cms-sitefinity/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/blog-starter&project-name=blog-starter&repository-name=blog-starter)
```
Comment (balazsorban44) [inline examples/cms-sitefinity/README.md:14]:
```suggestion
[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/cms-sitefinity&project-name=cms-sitefinity&repository-name=cms-sitefinity)
```

File: examples/cms-sitefinity/package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    "private": true,
+    "scripts": {
+        "dev": "next",
+        "build": "next build",
+        "start": "next start",
+        "typecheck": "tsc",
+        "lint-fix": "pnpm prettier --write . && eslint . --ext js,jsx,ts,tsx --fix --max-warnings=0 --config .eslintrc.json --no-eslintrc"
```
Comment (balazsorban44) [inline examples/cms-sitefinity/package.json:8]:
```suggestion
        "start": "next start"
```

The example should be minimal and focused.

File: examples/cms-sitefinity/pages/_document.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import Document, { Html, Head, Main, NextScript } from 'next/document'
```
Comment (balazsorban44) [inline examples/cms-sitefinity/pages/_document.tsx:1]: This does not do anything, can we just remove it?

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Added some small comments. Could you also explain what is `sitefinity/SitefinityExport.zip`? It does not seem to be referenced anywhere. Can we remove it?

File: examples/cms-sitefinity/pages/_document.tsx (hunk 4 above)
Comment (Gebov) [reply examples/cms-sitefinity/pages/_document.tsx:1]: I get errors without the import:

image

**Comment by Gebov** [comment]
@balazsorban44 The file sitefinity/SitefinityExport.zip is a CMS specific set of files that when imported into the CMS automatically create the necessary dynamic model for this example to work. An alternative to that file would be an update to the ReadMe with the manual steps to add each dynamic model in the Administration.

File: examples/cms-sitefinity/pages/_document.tsx (hunk 4 above)
Comment (balazsorban44) [reply examples/cms-sitefinity/pages/_document.tsx:1]: I meant the whole file.

**Comment by balazsorban44** [comment]
I'm sorry, now I see it in the readme: https://github.com/vercel/next.js/pull/39537/files#diff-7a26e518bdbb8b7ce72633518ef473e1b0dee52f1199e230a6182673da9699a7R65 :+1:

File: examples/cms-sitefinity/pages/_document.tsx (hunk 4 above)
Comment (balazsorban44) [reply examples/cms-sitefinity/pages/_document.tsx]: Removed it. :+1:

**Review by balazsorban44 — DISMISSED** [review]
Removed some files to keep it as simple as possible. Thanks for the PR!

File: examples/cms-sitefinity/package.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        "react-dom": "^18.2.0",
+        "typescript": "^4.7.4"
+    },
+    "devDependencies": {
+        "@types/node": "^18.0.3",
+        "@types/react": "^18.0.15",
+        "@types/react-dom": "^18.0.6",
+        "autoprefixer": "^10.4.7",
+        "postcss": "^8.4.14",
+        "tailwindcss": "^3.1.4"
```
Comment (balazsorban44) [inline examples/cms-sitefinity/package.json:22]:
```suggestion
        "react-dom": "^18.2.0"
    },
    "devDependencies": {
        "@types/node": "^18.0.3",
        "@types/react": "^18.0.15",
        "@types/react-dom": "^18.0.6",
        "autoprefixer": "^10.4.7",
        "postcss": "^8.4.14",
        "tailwindcss": "^3.1.4",
        "typescript": "^4.7.4"
```

**Review by balazsorban44 — DISMISSED** [review]

**Review by balazsorban44 — DISMISSED** [review]

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**